src/main/java/org/mule/extension/email/internal/mailbox/BaseMailboxPollingSource.java

```diff
@@ -196,21 +197,23 @@ protected void emailOnFlowError() {
 
   protected void beginUsingFolder() {
     synchronized (usingFolderCounter) {
-      LOGGER.debug("beginUsingFolder.");
+      LOGGER.debug("beginUsingFolder = " + usingFolderCounter.get());
```

Comment:
Please, when the log is not only text and you have to access to others methods, you should add an if to avoid processing it. Ex:
```
if (LOGGER.isDebugEnabled()) {
   LOGGER.debug("beginUsingFolder = " + usingFolderCounter.get());
}
```